gobblin-runtime/src/main/java/org/apache/gobblin/runtime/api/FlowSpecSearchObject.java

```diff
@@ -136,11 +147,11 @@ public String augmentBaseGetStatement(String baseStatement)
       }
     }
 
-    if (conditions.size() == 0) {
```

Comment:
Wouldn't an easier check be to check if limitAndOffset.size() == 0 ? I don't think you need an explicit boolean to determine when to get all in that case.